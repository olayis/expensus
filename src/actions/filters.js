// SET_TEXT_FILTER
export const setTextFilter = (text = '') => ({
  type: 'SET_TEXT_FILTER',
  text,
});

// SORT_BY_DATE_RECENT
export const sortByDateRecent = () => ({
  type: 'SORT_BY_DATE_RECENT',
});

// SORT_BY_DATE_OLDEST
export const sortByDateOldest = () => ({
  type: 'SORT_BY_DATE_OLDEST',
});

// SORT_BY_AMOUNT_HIGHEST
export const sortByAmountHighest = () => ({
  type: 'SORT_BY_AMOUNT_HIGHEST',
});

// SORT_BY_AMOUNT_LOWEST
export const sortByAmountLowest = () => ({
  type: 'SORT_BY_AMOUNT_LOWEST',
});

// SORT_BY_DESCRIPTION_A_TO_Z
export const sortByDescriptionAtoZ = () => ({
  type: 'SORT_BY_DESCRIPTION_A_TO_Z',
});

// SORT_BY_DESCRIPTION_Z_TO_A
export const sortByDescriptionZtoA = () => ({
  type: 'SORT_BY_DESCRIPTION_Z_TO_A',
});

// SORT_BY_CATEGORY
export const sortByCategory = () => ({
  type: 'SORT_BY_CATEGORY',
});

// SET_START_DATE
export const setStartDate = (startDate) => ({
  type: 'SET_START_DATE',
  startDate,
});

// SET_END_DATE
export const setEndDate = (endDate) => ({
  type: 'SET_END_DATE',
  endDate,
});
