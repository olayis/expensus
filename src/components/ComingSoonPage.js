import React from 'react';
import { Link } from 'react-router-dom';
import programming from '../../public/img/programming.svg';

const ComingSoonPage = () => (
  <main className='main-content'>
    <div className='content-container'>
      <div className='error'>
        <img
          src={programming}
          alt='Programmer'
          className='error__illustration'
          width='1075'
          height='585'
        />
        <h2 className='error__text'>This page is coming soon.</h2>
        <div className='error__explanation'>
          Continue managing your expenses.
          <span className='error__action'>
            <Link className='btn btn--try-again' to='/'>
              Go Home
            </Link>
          </span>
        </div>
      </div>
    </div>
  </main>
);

export default ComingSoonPage;
