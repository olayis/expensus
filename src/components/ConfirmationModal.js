import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => {
  if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app');

  return (
    <div>
      <Modal
        contentLabel='Are you sure?'
        isOpen={!!props.deleteSelected}
        onRequestClose={props.stopDeleteDataSelected}
        closeTimeoutMS={200}
        className='modal'
        aria={{
          labelledby: 'heading',
          describedby: 'full_description',
        }}
      >
        <h3 id='heading' className='modal__title'>
          Delete Data
        </h3>
        <p id='full_description' className='modal__body'>
          Are you sure you want to delete all of your data on Expensus?
        </p>
        <button className='modal__btn--delete' onClick={props.handleDeleteData}>
          Delete
        </button>
        <button
          className='modal__btn--cancel'
          onClick={props.stopDeleteDataSelected}
        >
          Cancel
        </button>
      </Modal>
    </div>
  );
};

export default ConfirmationModal;
