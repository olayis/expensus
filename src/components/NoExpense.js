import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import no_data from '../../public/img/no_data.svg';
import filter from '../../public/img/filter.svg';

export const NoExpense = (props) => (
  <div className='no-expense'>
    {props.expensesWithoutFilters.length === 0 ? (
      <>
        <img
          src={no_data}
          alt='A clipboard showing no data'
          className='no-expense__illustration'
          width='648'
          height='633'
        />
        <p>
          No Expenses To Display.{' '}
          <Link to='/create' className='router-link router-link--add'>
            Add an Expense
          </Link>
        </p>
      </>
    ) : (
      <>
        <img
          src={filter}
          alt='A filter showing Expenses have been filtered'
          className='no-expense__illustration'
          width='883'
          height='608'
        />
        <p>
          No results found {props.filters.text && `for "${props.filters.text}"`}{' '}
          {props.filters.startDate &&
            `starting from ${moment(props.filters.startDate).format(
              'MMM Do, YYYY'
            )}`}{' '}
          {props.filters.endDate &&
            `ending at ${moment(props.filters.endDate).format('MMM Do, YYYY')}`}
        </p>
      </>
    )}
  </div>
);

const mapStateToProps = (state) => {
  return {
    filters: state.filters,
  };
};

export default connect(mapStateToProps)(NoExpense);
