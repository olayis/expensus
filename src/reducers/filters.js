// Filters Reducer
import moment from 'moment';

const filtersReducerDefaultState = {
  text: '',
  sortBy: 'date(recent)',
  startDate: moment().startOf('month'),
  endDate: moment().endOf('month'),
};

export default (state = filtersReducerDefaultState, action) => {
  switch (action.type) {
    case 'SET_TEXT_FILTER':
      return {
        ...state,
        text: action.text,
      };

    case 'SORT_BY_DATE_RECENT':
      return {
        ...state,
        sortBy: 'date(recent)',
      };

    case 'SORT_BY_DATE_OLDEST':
      return {
        ...state,
        sortBy: 'date(oldest)',
      };

    case 'SORT_BY_AMOUNT_HIGHEST':
      return {
        ...state,
        sortBy: 'amount(highest)',
      };

    case 'SORT_BY_AMOUNT_LOWEST':
      return {
        ...state,
        sortBy: 'amount(lowest)',
      };

    case 'SORT_BY_DESCRIPTION_A_TO_Z':
      return {
        ...state,
        sortBy: 'description(a_to_z)',
      };

    case 'SORT_BY_DESCRIPTION_Z_TO_A':
      return {
        ...state,
        sortBy: 'description(z_to_a)',
      };

    case 'SORT_BY_CATEGORY':
      return {
        ...state,
        sortBy: 'category',
      };

    case 'SET_START_DATE':
      return {
        ...state,
        startDate: action.startDate,
      };

    case 'SET_END_DATE':
      return {
        ...state,
        endDate: action.endDate,
      };

    default:
      return state;
  }
};
