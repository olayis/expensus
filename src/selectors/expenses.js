// Get Visible expenses
import moment from 'moment';

export default (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses
    .filter((expense) => {
      const createdAtMoment = moment(expense.createdAt);
      const startDateMatch = startDate
        ? startDate.isSameOrBefore(createdAtMoment, 'day')
        : true;
      const endDateMatch = endDate
        ? endDate.isSameOrAfter(createdAtMoment, 'day')
        : true;
      const textMatch = expense.description
        .toLowerCase()
        .includes(text.toLowerCase());

      return textMatch && startDateMatch && endDateMatch;
    })
    .sort((a, b) => {
      switch (sortBy) {
        case 'date(recent)':
          return a.createdAt < b.createdAt ? 1 : -1;

        case 'date(oldest)':
          return a.createdAt > b.createdAt ? 1 : -1;

        case 'amount(highest)':
          return a.amount < b.amount ? 1 : -1;

        case 'amount(lowest)':
          return a.amount > b.amount ? 1 : -1;

        case 'description(a_to_z)':
          return a.description.localeCompare(b.description);

        case 'description(z_to_a)':
          return b.description.localeCompare(a.description);

        case 'category':
          return a.category.localeCompare(b.category);
      }
    });
};
