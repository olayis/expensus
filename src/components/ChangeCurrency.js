import React from 'react';
import { connect } from 'react-redux';
import { startChangeLocale } from '../actions/locale';
import CurrencyForm from './CurrencyForm';

export class ChangeCurrency extends React.Component {
  onSubmit = ({ currency }) => {
    this.props.startChangeLocale({ currency });
    this.props.history.push('/');
    this.props.history.go(0);
  };

  goBack = () => {
    this.props.history.goBack();
  };

  render() {
    return (
      <main className='main-content'>
        <div className='content-container'>
          <div className='component-container component-container--currency'>
            <CurrencyForm
              onSubmit={this.onSubmit}
              currency={this.props.currency}
              goBack={this.goBack}
            />
          </div>
        </div>
      </main>
    );
  }
}

const mapStateToProps = (state) => ({
  currency: state.locale ? state.locale.currency : '',
});

const mapDispatchToProps = (dispatch) => ({
  startChangeLocale: (currency) => dispatch(startChangeLocale(currency)),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChangeCurrency);
