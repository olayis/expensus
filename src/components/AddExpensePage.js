import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';
import { startChangeCurrency } from '../actions/locale';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.startChangeCurrency({ currency: 'Danish Krone' });
    this.props.history.push('/');
  };
  render() {
    return (
      <main className='main-content'>
        <div className='content-container'>
          <div className='add-expense-container'>
            <ExpenseForm onSubmit={this.onSubmit} />
          </div>
        </div>
      </main>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => dispatch(startAddExpense(expense)),
  startChangeCurrency: (currency) => dispatch(startChangeCurrency(currency)),
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
