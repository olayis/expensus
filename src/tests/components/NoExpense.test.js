import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import { NoExpense } from '../../components/NoExpense';
import { altFilters } from '../fixtures/filters';

test('should render NoExpense with empty message', () => {
  const wrapper = shallow(<NoExpense expensesWithoutFilters={[]} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render NoExpense with text and date filters message', () => {
  const wrapper = shallow(
    <NoExpense expensesWithoutFilters={expenses} filters={altFilters} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render NoExpense with text filter message', () => {
  const filters = {
    ...altFilters,
    startDate: null,
    endDate: null,
  };
  const wrapper = shallow(
    <NoExpense expensesWithoutFilters={expenses} filters={filters} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render NoExpense with dates filter message', () => {
  const filters = {
    ...altFilters,
    text: '',
  };
  const wrapper = shallow(
    <NoExpense expensesWithoutFilters={expenses} filters={filters} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render NoExpense with startDate filter message', () => {
  const filters = {
    ...altFilters,
    text: '',
    endDate: null,
  };
  const wrapper = shallow(
    <NoExpense expensesWithoutFilters={expenses} filters={filters} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render NoExpense with endDate filter message', () => {
  const filters = {
    ...altFilters,
    text: '',
    startDate: null,
  };
  const wrapper = shallow(
    <NoExpense expensesWithoutFilters={expenses} filters={filters} />
  );
  expect(wrapper).toMatchSnapshot();
});
