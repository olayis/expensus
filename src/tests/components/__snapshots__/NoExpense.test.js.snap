// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NoExpense with dates filter message 1`] = `
<div
  className="no-expense"
>
  <img
    alt="A filter showing Expenses have been filtered"
    className="no-expense__illustration"
    height="608"
    src="SvgrURL"
    width="883"
  />
  <p>
    No results found 
     
    starting from Jan 1st, 1970
     
    ending at Jan 5th, 1970
  </p>
</div>
`;

exports[`should render NoExpense with empty message 1`] = `
<div
  className="no-expense"
>
  <img
    alt="A clipboard showing no data"
    className="no-expense__illustration"
    height="633"
    src="SvgrURL"
    width="648"
  />
  <p>
    No Expenses To Display.
     
    <Link
      className="router-link router-link--add"
      to="/create"
    >
      Add an Expense
    </Link>
  </p>
</div>
`;

exports[`should render NoExpense with endDate filter message 1`] = `
<div
  className="no-expense"
>
  <img
    alt="A filter showing Expenses have been filtered"
    className="no-expense__illustration"
    height="608"
    src="SvgrURL"
    width="883"
  />
  <p>
    No results found 
     
     
    ending at Jan 5th, 1970
  </p>
</div>
`;

exports[`should render NoExpense with startDate filter message 1`] = `
<div
  className="no-expense"
>
  <img
    alt="A filter showing Expenses have been filtered"
    className="no-expense__illustration"
    height="608"
    src="SvgrURL"
    width="883"
  />
  <p>
    No results found 
     
    starting from Jan 1st, 1970
     
  </p>
</div>
`;

exports[`should render NoExpense with text and date filters message 1`] = `
<div
  className="no-expense"
>
  <img
    alt="A filter showing Expenses have been filtered"
    className="no-expense__illustration"
    height="608"
    src="SvgrURL"
    width="883"
  />
  <p>
    No results found 
    for "bills"
     
    starting from Jan 1st, 1970
     
    ending at Jan 5th, 1970
  </p>
</div>
`;

exports[`should render NoExpense with text filter message 1`] = `
<div
  className="no-expense"
>
  <img
    alt="A filter showing Expenses have been filtered"
    className="no-expense__illustration"
    height="608"
    src="SvgrURL"
    width="883"
  />
  <p>
    No results found 
    for "bills"
     
     
  </p>
</div>
`;
